components/content_settings/renderer/brave_content_settings_agent_impl_browsertest.cc

```diff
@@ -748,20 +746,18 @@ IN_PROC_BROWSER_TEST_F(BraveContentSettingsAgentImplBrowserTest,
       ExecScriptGetStr(create_image_script(redirect_to_cross_site_image_url()),
                        contents()),
       redirect_to_cross_site_image_url().spec());
-  EXPECT_EQ(GetLastReferrer(cross_site_image_url()),
-            cross_site_url().GetOrigin().spec());
+  EXPECT_EQ(GetLastReferrer(cross_site_image_url()), url().GetOrigin().spec());
 
   // Cross-site iframe navigations should follow the default referrer policy.
   NavigateCrossSiteRedirectIframe();
   EXPECT_EQ(ExecScriptGetStr(kReferrerScript, child_frame()),
-            cross_site_url().GetOrigin().spec());
-  EXPECT_EQ(GetLastReferrer(redirect_to_cross_site_url()),
-            cross_site_url().GetOrigin().spec());
+            url().GetOrigin().spec());
+  EXPECT_EQ(GetLastReferrer(cross_site_url()), url().GetOrigin().spec());
 
   // Cross-site navigations get no referrer.
   RedirectToPageWithLink(redirect_to_cross_site_url(), cross_site_url());
   EXPECT_EQ(ExecScriptGetStr(kReferrerScript, contents()), "");
-  EXPECT_EQ(GetLastReferrer(redirect_to_cross_site_url()), "");
```

Comment:
Same as my previous comment, we should be able to check that the referrer of the intermediate steps are also correctly set.

Reply:
added